### PR TITLE
finished 12 challenge and added a README.md file and whiteboard

### PR DESCRIPTION
**Pull request by GoldenDog190**

### CODE REVIEW

**Comment by GoldenDog190** [comment]
[x ] Top-level README “Table of Contents” is updated
 - [ x] Feature tasks for this challenge are completed
 - [x ] Unit tests written and passing
     - [ x] “Happy Path” - Expected outcome
     - [x ] Expected failure
     - [ x] Edge Case (if applicable/obvious)
 - [ x] README for this challenge is complete
     - [ x] Summary, Description, Approach & Efficiency, Solution
     - [ x] Link to code
     - [ x] Picture of whiteboard

File: challenges/lib/src/main/java/challenges/utilities/AnimalShelter.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import challenges.stacksandqueues.Queue;
+
+import java.util.Locale;
+
+public class AnimalShelter<T> {
+    Queue dogs = new Queue<T>();
+    Queue cats = new Queue<T>();
+
+    public void enqueue(T animal) throws ClassNotFoundException {
```
Comment (Mattpet26) [inline challenges/lib/src/main/java/challenges/utilities/AnimalShelter.java:11]: Good work on making it accept a generic animal, instead of writing out two separate methods!

File: challenges/lib/src/test/java/challenges/utilities/AnimalShelterTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    Cat catTest3 = new Cat("Lucy3");
+
+    testAnimalShelter.enqueue(dogTest);
+    testAnimalShelter.enqueue(catTest);
+    testAnimalShelter.enqueue(dogTest2);
+    testAnimalShelter.enqueue(catTest2);
+    testAnimalShelter.enqueue(dogTest3);
+    testAnimalShelter.enqueue(catTest3);
+
+    System.out.println(testAnimalShelter.dequeue("DOG"));
```
Comment (Mattpet26) [inline challenges/lib/src/test/java/challenges/utilities/AnimalShelterTest.java:22]: system.out.println is not a test.
